### ISSUE TITLE
Initialize project structure with Hydra configuration system and development tooling

### ISSUE THREAD

**Issue by chrisjcc** [chrisjcc/text-to-sql-refine-tuning#4]
**Assignee:** @claude
**Labels:** setup, infrastructure, configuration

**Description:**
Set up the foundational project structure including the Hydra-based configuration system, logging infrastructure, dependency management, and development tooling.

**Requirements**
#### 1. Project Directory Structure
Create the following directory structure:
```
text-to-sql-fine-tuning/
├── .env.example
├── .gitignore
├── pyproject.toml
├── requirements.txt
├── setup.py
├── README.md
├── config/
│   ├── __init__.py
│   ├── config.py
│   ├── config.yaml
│   ├── dataset/
│   │   └── dataset.yaml
│   ├── evaluation/
│   │   └── evaluation.yaml
│   ├── hf/
│   │   └── hf.yaml
│   ├── inference/
│   │   └── inference.yaml
│   ├── training/
│   │   └── training.yaml
│   └── wandb/
│       └── wandb.yaml
├── src/
│   ├── __init__.py
│   ├── utils/
│   │   ├── __init__.py
│   │   └── logging_utils.py
│   └── environments/
│       └── __init__.py
├── scripts/
│   └── __init__.py
├── logs/
│   └── .gitkeep
└── tests/
    └── __init__.py
```

#### 2. Configuration Files
config/config.yaml:
```
defaults:
  - hf: hf
  - dataset: dataset
  - wandb: wandb
  - training: training
  - evaluation: evaluation
  - inference: inference
  - _self_

logging:
  level: INFO
  log_dir: logs
  format: "%(asctime)s - %(name)s - %(levelname)s - %(message)s"

project:
  name: text-to-sql-fine-tuning
  version: 0.1.0
  seed: 42
```

config/hf/hf.yaml:
```
model:
  name: meta-llama/Llama-3.1-8B-Instruct
  cache_dir: ./cache
  trust_remote_code: true

hub:
  push_to_hub: false
  repo_name: null
  private: true
```

config/dataset/dataset.yaml:
```
name: b-mc2/sql-create-context
split:
  train: train
  validation: validation
  test: null

preprocessing:
  max_length: 512
  truncation: true

cache_dir: ./data_cache
```

config/training/training.yaml:
```
# GRPO Training Configuration
output_dir: ./outputs
run_name: grpo-text-to-sql

# Model & LoRA
use_peft: true
peft:
  lora_r: 16
  lora_alpha: 32
  lora_dropout: 0.05
  target_modules: ["q_proj", "v_proj", "k_proj", "o_proj"]
  use_qlora: true
  bnb_4bit_compute_dtype: bfloat16

# Batch Configuration
per_device_train_batch_size: 1
gradient_accumulation_steps: 8
num_train_epochs: 3
max_steps: -1

# Learning Rate & Schedule
learning_rate: 5.0e-6
lr_scheduler_type: cosine
warmup_steps: 100
warmup_ratio: 0.03

# Generation Parameters
max_new_tokens: 256
temperature: 0.7
top_p: 0.9
do_sample: true

# GRPO-Specific
num_generations: 4
kl_coef: 0.05
gamma: 1.0

# Optimization
bf16: true
gradient_checkpointing: true
optim: adamw_torch

# Saving & Logging
save_strategy: steps
save_steps: 500
save_total_limit: 3
logging_steps: 10

# Evaluation
evaluation_strategy: steps
eval_steps: 500
```

config/evaluation/evaluation.yaml:
```
metrics:
  - valid_sql_percentage
  - execution_accuracy
  - exact_match

rubric:
  sql_keywords: ["SELECT", "FROM", "WHERE", "JOIN", "INSERT", "UPDATE", "DELETE", "CREATE"]
  check_syntax: true
  check_executability: false  # Requires DB connection

batch_size: 8
```

config/wandb/wandb.yaml:
```
enabled: true
project: text-to-sql-grpo
entity: null  # Your wandb username
name: null  # Auto-generated from run_name
log_model: false
```

config/inference/inference.yaml:
```
batch_size: 4
max_new_tokens: 256
temperature: 0.1
top_p: 0.95
num_beams: 1
do_sample: false
```

#### 3. Configuration Loader (config/config.py)

Implement the config loader with proper error handling and environment variable loading as specified in your document.

#### 4. Logging Utility (src/utils/logging_utils.py)

Create a logging utility that:
- Sets up structured logging with the format from config
- Creates log files in the logs directory
- Supports different log levels
- Integrates with Hydra's logging system

#### 5. Dependency Management

**requirements.txt** should include:
```
# Core
torch>=2.1.0
transformers>=4.36.0
datasets>=2.14.0
accelerate>=0.25.0
trl>=0.7.0
peft>=0.7.0
bitsandbytes>=0.41.0

# Verifiers
verifiers>=0.1.0

# Configuration & Logging
hydra-core>=1.3.0
omegaconf>=2.3.0
python-dotenv>=1.0.0
wandb>=0.16.0

# SQL & Evaluation
sqlparse>=0.4.4

# Utilities
numpy>=1.24.0
pandas>=2.0.0
tqdm>=4.66.0
```

#### 6. Environment Variables (.env.example)
```
# HuggingFace
HF_TOKEN=your_huggingface_token_here
HF_HOME=./cache

# Weights & Biases
WANDB_API_KEY=your_wandb_key_here
WANDB_PROJECT=text-to-sql-grpo

# Training
CUDA_VISIBLE_DEVICES=0
```
#### 7. .gitignore
Include standard Python ignores plus:
```
.env
logs/
outputs/
cache/
data_cache/
wandb/
*.pt, *.bin, *.safetensors (model checkpoints)
```

**Acceptance Criteria**
- [x] All directories are created with proper __init__.py files
- [x]  All Hydra config files are properly structured and load without errors
- [x]  config.py successfully loads and merges all config files
- [x]  Logging utility creates logs in the correct format and location
- [x]  .env.example is created (but .env is gitignored)
- [x]  requirements.txt includes all necessary dependencies
- [x]  .gitignore prevents committing sensitive and generated files
- [x]  Project can be installed with pip install -r requirements.txt
- [x]  Running python config/config.py successfully loads and displays config

**Testing**
Run the following to verify setup:
```python
python config/config.py
```

Expected output should show loaded configuration without errors.
